internal/state/change_processor_test.go

```diff
@@ -316,18 +316,20 @@ var _ = Describe("ChangeProcessor", func() {
 								ObservedGeneration: gw1.Generation,
 								ListenerStatuses: map[string]state.ListenerStatus{
 									"listener-80-1": {
-										AttachedRoutes: 1,
-										Conditions: append(
-											conditions.NewDefaultListenerConditions(),
-											conditions.NewTODO("GatewayClass is invalid or doesn't exist"),
-										),
+										AttachedRoutes: 0,
```

Comment:
HTTPRoutes do not longer attach when GatewayClass doesn't exist or invalid, because the listeners become invalid.